### PR TITLE
Search / Filter groups by group name

### LINKED ISSUE #386: Search / Filter groups by group name (40-80 Subs)

### PR DESCRIPTION
**Pull request by TripleSpeeder**
This fixes #386

### CODE REVIEW

File: BrightID/src/components/GroupsScreens/GroupsScreen.js [hunk 1]
```diff
@@ -128,4 +157,26 @@ const styles = StyleSheet.create({
...
+  const hasGroups = groups.length > 0;
+
+  // apply search filter to groups array
+  // NOTE: If below sorting/filtering gets too expensive at runtime use memoized selectors / reselect
+  if (searchParam !== '') {
+    groups = groups.filter((group) => {
+      if (group.type === PRIMARY_GROUP_TYPE) {
+        // since primary group does not have a name, match against the default name 'primary Group'
+        // this allows the user to explicitly search for the primary group
+        return PRIMARY_GROUP_NAME.includes(searchParam);
```
Comment (TripleSpeeder) [inline BrightID/src/components/GroupsScreens/GroupsScreen.js:172]: Had the wrong assumption here that primary groups don't have a name. Just found out they do! Will update PR with a fix :)

**Comment by TripleSpeeder** [comment]
This is now rebased to current dev.

**Comment by RnbWd** [comment]
the code looks pretty good, but I'd also like to add a feature for searching for the old groups names if the group doesn't have a name. There's a helper function for this, import { getGroupName } from '@/utils/groups';

**Comment by adamstallard** [comment]
> the code looks pretty good, but I'd also like to add a feature for searching for the old groups names if the group doesn't have a name. There's a helper function for this, import { getGroupName } from '@/utils/groups';

Would that also solve this issue? https://github.com/BrightID/BrightID/issues/273 ("Search / Filter groups by name of user in group")

**Comment by TripleSpeeder** [comment]
> the code looks pretty good, but I'd also like to add a feature for searching for the old groups names if the group doesn't have a name. There's a helper function for this, import { getGroupName } from '@/utils/groups';

Good! Just updated the PR to use this.

> Would that also solve this issue? #273 ("Search / Filter groups by name of user in group")

Don't think so. `getGroupName()` only returns 3 members, and their names are cut of at 13 characters.

**Pull request merged**